src/System.Memory/tests/Performance/Perf.Span.IndexOf.cs

```diff
@@ -163,5 +163,25 @@ public void StringLastIndexOfChar(int size)
             }
             Assert.Equal(size / 2, index);
         }
+
+        [InlineData("foo", "", StringComparison.Ordinal)]
+        [InlineData("foobardzsdzs", "rddzs", StringComparison.Ordinal)]
+        [InlineData("Hello", "L", StringComparison.OrdinalIgnoreCase)]
+        [InlineData("Exhibit \u00C0", "a\u0300", StringComparison.CurrentCultureIgnoreCase)]
+        [InlineData("Exhibit \u00C0", "a\u0300", StringComparison.OrdinalIgnoreCase)]
+        [InlineData("TestFooBA\u0300R", "FooB\u00C0R", StringComparison.InvariantCultureIgnoreCase)]
+        [InlineData("More Test's", "Tests", StringComparison.InvariantCulture)]
+        [InlineData("very long input string that is hardly recognizable by a clever PC and therefore not usable", "PC", StringComparison.Ordinal)]
+        public void SpanIndexOfSpanComparison(string input, string value, StringComparison comparisonType)
+        {
+            ReadOnlySpan<char> inputSpan = input.AsSpan();
+            ReadOnlySpan<char> valueSpan = value.AsSpan();
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                {
+                    inputSpan.IndexOf(valueSpan, comparisonType);
```

Comment:
@eerhardt had some tests he used when SIMD'ifying IndexOf - I know he used long strings with and without matches, because for shorter strings SIMD was slower.

Reply:
Here were the tests I used at the time:

https://gist.github.com/eerhardt/cbfdb7cb0d53cf85d5d220555de8cb4c